src/components/data-exploration/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -91,21 +79,105 @@ const DiffExprCompute = (props) => {
       comparisonGroup[selectedComparison]['basis'] = `sample/${samples[0].key}`
     }
 
+    setSampleKeys(samples.map(sample => sample.key));
+
   }, [hierarchy, properties]);
 
-  const validateForm = () => {
+  const cellIdToSampleMap = useMemo(() => {
+    const mapping = [];
+    sampleKeys.forEach((key, idx) => {
+      const cellIds = properties[key].cellIds;
+      cellIds.forEach(cellId => mapping[cellId] = idx);
+    });
 
-    if (!comparisonGroup[selectedComparison]?.cellSet) {
-      setIsFormValid(false);
-      return;
+    return mapping;
+  }, [numSamples]);
+
+  const canRunDiffExpr = useCallback(() => {
+
+    if (selectedComparison === ComparisonType.WITHIN) return true;
+
+    if (!basis || !cellSet || !compareWith || !cellIdToSampleMap.length > 0) { return false; }
+
+    const basisCellSetKey = getCellSetName(basis);
+
+    // Group 1 is from cellSet
+    // Group 2 is from compareWith
```

Comment:
instead of adding a comment for this, why don't you just call them `cellSetKey` and `compareWithKey`?

Reply:
Hmm.. ok. Changed.